frontend/context/StakingContractDetailsProvider.tsx

```diff
@@ -0,0 +1,181 @@
+import { useQueries, useQuery } from '@tanstack/react-query';
+import {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useCallback,
+  useContext,
+  useState,
+} from 'react';
+
+import { AGENT_CONFIG } from '@/config/agents';
+import { GNOSIS_CHAIN_CONFIG } from '@/config/chains';
+import { FIVE_SECONDS_INTERVAL } from '@/constants/intervals';
+import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
+import { StakingProgramId } from '@/enums/StakingProgram';
+import { useService } from '@/hooks/useService';
+import { useServices } from '@/hooks/useServices';
+import { StakingContractDetails } from '@/types/Autonolas';
+
+import {
+  INITIAL_DEFAULT_STAKING_PROGRAM_ID,
+  StakingProgramContext,
+} from './StakingProgramProvider';
+
+const currentAgent = AGENT_CONFIG.trader; // TODO: replace with dynamic agent selection
+const currentChainId = GNOSIS_CHAIN_CONFIG.chainId; // TODO: replace with dynamic chain selection
+
+// TODO: create a separate hook to get service id
+const useServiceId = () => {
+  const { selectedService, isFetched: isLoaded } = useServices();
+  const serviceConfigId =
+    isLoaded && selectedService ? selectedService?.service_config_id : '';
+  const { service } = useService({ serviceConfigId });
+  const serviceId = service?.chain_configs[currentChainId].chain_data?.token;
+
+  return serviceId;
+};
+
+/**
+ * hook to get staking contract details by staking program
+ */
+const useStakingContractDetailsByStakingProgram = (isPaused: boolean) => {
+  const { activeStakingProgramId } = useContext(StakingProgramContext);
+  const serviceId = useServiceId();
```

Comment:
couple things here..

ideally this could be moved to hooks and exported, following the current pattern, and probably accept a param of `stakingProgramId`, so we can pull data by staking program id, regardless of whether it is actively staked

the name of the hook does not highlight that it's only fetching the `activeStakingProgramId`